File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/03_Conditional_Statements_Advanced_Exercise/09_Ski_Trip.js
function skiTrip(input) {
    let roomForOne = 18;
    let apartment = 25;
    let presidentApartment = 35;
    
    let days = Number(input[0]);
    let nigths = days - 1;
    let roomType = input[1];
    let happy = input[2];  // "positive"  или "negative"
    
    let roomPrice = 0;

    if (roomType == "room for one person") {
        roomPrice = roomForOne;
    } else if (roomType == "apartment") {
        roomPrice = apartment;
    } else if (roomType == "president apartment") {
        roomPrice = presidentApartment;
    }

    let totalCost = roomPrice * nigths;
    if (roomType == "apartment") {
        if (nigths < 10) {
            totalCost *= 0.70;
        } else if (nigths <= 15) {
            totalCost *= 0.65;
        } else if (nigths > 15) {
            totalCost *= 0.50;
        }

    } else if (roomType == "president apartment") {
        if (nigths < 10) {
            totalCost *= 0.90;
        } else if (nigths <= 15) {
            totalCost *= 0.85;
        } else if (nigths > 15) {
            totalCost *= 0.80;
        }
    }

    if (happy == "positive") {
        totalCost *= 1.25;
    } else if (happy == "negative") {
        totalCost *= 0.90;
    }

    console.log(totalCost.toFixed(2));
}


// skiTrip(["14", "apartment", "positive"]);

File: Python_Courses/Python_Basics/Python_PB_2024_01/03_Conditional_Statements_Advanced_Exercise/09_Ski_Trip.py
days = int(input())
stay_type = input()  # "room for one person", "apartment", or "president apartment"
evaluation = input()  # "positive" or "negative"

# nights = days - 1
cost_per_night = 0
discount = 0
if stay_type == 'room for one person':
    cost_per_night = 18.00
elif stay_type == 'apartment':
    cost_per_night = 25.00
    if days < 10:
        discount = 0.30
    elif days <= 15:  # [10;15]
        discount = 0.35
    else:
        discount = 0.50
elif stay_type == 'president apartment':
    cost_per_night = 35.00
    if days < 10:
        discount = 0.10
    elif days <= 15:  # [10;15]
        discount = 0.15
    else:
        discount = 0.20

total_cost = (days - 1) * cost_per_night * (1 - discount)

if evaluation == 'positive':
    total_cost *= 1.25  # total_cost = total_cost * (1 + 0.25)
elif evaluation == 'negative':
    total_cost *= 0.90  # total_cost = total_cost * (1 - 0.10)

print(f'{total_cost :.2f}')
